### ISSUE TITLE
Improve html examples

### ISSUE THREAD

**Issue by equinusocio** [bigfishtv/turret#28]
I think some html examples should be improved to add more semantic with appropriate tags and where required, roles and aria-attributes.

For example, https://turretcss.com/form/field/

The field element should be by semantic a `<fieldset>` not a div..

And this example does not make sense since `<fieldset>` are made to group parts of form elements, not to wrap all the elements inside the form.
https://turretcss.com/form/fieldset/

Btw i think also the fieldset style must be reviewed.

**Comment by scottdejonge** [comment]
Hey @equinusocio I agree that the `<fieldset>` styles should be reviewed (or removed entirely) to provide more flexibility and use of `<fieldset>` to wrap `<input>` elements.

> The field element should be by semantic a `<fieldset>` not a div.

I am not sure that the above statement, is correct. Yes, `<fieldset>` should be used to wrap like options (radio, checkbox) that are relevant to eachother, otherwise it's semantically appropriate to use a wrapping `<div>` or `<p>` to wrap a `<label>` and `<input>`.

Source: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset

Using `<feildset>` for this purpose (purely to wrap and control layout/margin) may cause semantic issues by grouping `<input>` that are not relevant together.

This is my understanding, I am open to your interpretation, and what your views are of how you would like `<feildset>` to be styled.

**Comment by equinusocio** [comment]
As per specification, the `fieldset` is made to group elements inside a form, it might contains any kind of elements and was not made only for grouping radios input. They can be also nested. Check the examples from the living specification
https://html.spec.whatwg.org/multipage/form-elements.html#the-fieldset-element

> The fieldset element represents a set of form controls (or other content) grouped together, optionally with a caption. 

As you can see it must be used to wrap related elements inside a group like label + input. Here a live valid example. It can also accept 3 useful attributes to disable an entire group, define a name or link a group of inputs outside the form to a specific form.

here a live demo
https://jsfiddle.net/bp6k8meh/2/

**Comment by scottdejonge** [comment]
Interesting! I haven't seen nested `<fieldset>` tags before.

Although in the [HTML Spec](https://html.spec.whatwg.org/multipage/form-elements.html#the-fieldset-element) for the `<fieldset>`element they are still using `<p>` tags to wrap `<label>`/ `<input>` pairs.

It's clear that the current implementation of `<fieldset>` styles are not working and need to be rethought.

**Comment by scottdejonge** [comment]
Default property definitions for `<fieldset>` has been greatly simplified in [v5.2.0](https://github.com/turretcss/turretcss/releases/tag/v5.2.0) to reduce opinionated styles.

Doc examples have been updated to use `<p>` tags to wrap `<label>` and `<input>` pairs.
